libcontainerd/remote/client.go

```diff
@@ -569,7 +569,7 @@ func (c *client) CreateCheckpoint(ctx context.Context, containerID, checkpointDi
 	var cpDesc *v1.Descriptor
 	for _, m := range index.Manifests {
 		if m.MediaType == images.MediaTypeContainerd1Checkpoint {
-			cpDesc = &m
+			cpDesc = &m // nolint:gosec
```

Comment:
I think this one may be legit, and should probably be:

```suggestion
	for _, m := range index.Manifests {
		m := m
		if m.MediaType == images.MediaTypeContainerd1Checkpoint {
			cpDesc = &m
```

Reply:
The `break` on the next line makes me think this is irrelevant.  The warning is that `m` will have the same address through every iteration of the loop and taking the address could mean reading a different value than you intend; however, since we're immediately ending the loop and no new data will be written into `m` we have the correct value in `cpDesc`.

The `m := m` suggestion should silence the warning and be resilient to further changes in this function so it might be slightly better from a maintenance perspective, but I'm not really concerned with the code as-is.